File: BE-FindingJob/src/main/java/com/example/befindingjob/service/UserService.java
package com.example.befindingjob.service;

import com.example.befindingjob.dto.UserDTO;
import com.example.befindingjob.dto.auth.*;
import com.example.befindingjob.entity.User;
import com.example.befindingjob.entity.enumm.Role;
import com.example.befindingjob.model.ApiResponse;

public interface UserService {
    ApiResponse<Void> register(Register_ResetPwdRequest registerRequest);
    ApiResponse<Void> otpRegister(OtpRequest otpRequest);
    ApiResponse<LoginResponse> login(LoginRequest loginRequest);
    ApiResponse<Void> forgetPwdRequest(OtpRequest otpRequest);
    ApiResponse<Void> verifyOtpRequest(OtpVerifyRequest otpVerifyResponse);
    ApiResponse<Void> passwordReset(Register_ResetPwdRequest resetPwdRequest);
    ApiResponse<Void> updateUser(UserDTO userDTO);

    ApiResponse<UserDTO> getUser(String token);
    ApiResponse<UserDTO> deleteAccountUser(String token);
    ApiResponse<UserDTO> findByEmail(String email);
}

File: JobHub/app/src/main/java/com/example/jobhub/service/UserService.kt
package com.example.jobhub.service

import com.example.jobhub.dto.UserDTO
import com.example.jobhub.dto.auth.OtpRequest
import com.example.jobhub.dto.auth.LoginRequest
import com.example.jobhub.dto.auth.LoginResponse
import com.example.jobhub.dto.auth.OtpVerifyRequest
import com.example.jobhub.dto.auth.Register_ResetPwdRequest
import com.example.jobhub.entity.enumm.Role
import com.example.jobhub.model.ApiResponse
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.POST
import retrofit2.http.Path

interface UserService {
    @POST("api/auth/register")
    fun register(@Body registerRequest: Register_ResetPwdRequest): Call<ApiResponse<Void>>

    @POST("api/auth/otp-register")
    fun otpRegister(@Body otpRequest: OtpRequest): Call<ApiResponse<Void>>

    @POST("api/auth/login")
    fun login(@Body loginRequest: LoginRequest): Call<ApiResponse<LoginResponse>>

    @POST("api/auth/request")
    fun resetPasswordRequest(@Body otpRequest: OtpRequest): Call<ApiResponse<Void>>

    @POST("api/auth/verify")
    fun verifyOtp(@Body otpVerifyRequest: OtpVerifyRequest): Call<ApiResponse<Void>>

    @POST("api/auth/reset")
    fun passwordReset(@Body resetPwdRequest: Register_ResetPwdRequest): Call<ApiResponse<Void>>

    @POST("api/users/update-user")
    fun updateUser(@Body userDTO: UserDTO): Call<ApiResponse<Void>>

    @GET("api/users/me")
    fun getUser(@Header("token") token: String): Call<ApiResponse<UserDTO>>

    @GET("api/users/find-by-email")
    fun findByEmail(@Header("email") email: String): Call<ApiResponse<UserDTO>>

    @DELETE("api/users/delete-account")
    fun deleteAccount(@Header("token") token: String): Call<ApiResponse<UserDTO>>

}